app/src/main/java/com/netcetera/skopjepulse/map/MapFragment.kt

```diff
@@ -269,9 +268,11 @@ class MapFragment : BaseFragment<MapViewModel>() {
   private fun setDaysNames() {
     val cal = Calendar.getInstance()
     val listOfDaysNames = listOf(nameOneDayAgo, nameTwoDaysAgo, nameThreeDaysAgo, nameFourDaysAgo, nameFiveDaysAgo, nameSixDaysAgo, nameSevenDaysAgo)
-    listOfDaysNames.forEach{
+    listOfDaysNames.forEach {
       cal.add(Calendar.DATE, -1)
-      it.text = SimpleDateFormat("EEE", Locale.ENGLISH).format(cal.time)
+      val language = context?.getSharedPreferences(Constants.LANGUAGE_CODE, Context.MODE_PRIVATE)?.getString(Constants.LANGUAGE_CODE, "en")
+        ?: "en"
+      it.text = SimpleDateFormat("EEE", Locale(language)).format(cal.time)
```

Comment:
Move the language extraction and the creation of the date formatter outside the loop